qiita_ware/ebi.py

```diff
@@ -143,33 +148,72 @@ def __init__(self, preprocessed_data_id, action):
         else:
             # This should never happen
-            raise ValueError("Unrecognized investigation type: '%s'. This "
-                             "term is neither one of the official terms nor "
-                             "one of the user-defined terms in the ENA "
-                             "ontology")
-
-        self.ebi_dir = join(qiita_config.working_dir, 'ebi_submission_',
-                            str(preprocessed_data_id))
-        self.sequence_files = []
-        self.study_xml_fp = None
-        self.sample_xml_fp = None
-        self.experiment_xml_fp = None
-        self.run_xml_fp = None
-        self.submission_xml_fp = None
+            error_msgs.append("Unrecognized investigation type: '%s'. This "
+                              "term is neither one of the official terms nor "
+                              "one of the user-defined terms in the ENA "
+                              "ontology." % it)
+
+        self.ebi_dir = join(qiita_config.working_dir,
+                            'ebi_submission_%d' % preprocessed_data_id)
+        get_output_fp = partial(join, self.ebi_dir, 'xml_dir')
+        self.xml_dir = get_output_fp()
+        self.study_xml_fp = get_output_fp('study.xml')
+        self.sample_xml_fp = get_output_fp('sample.xml')
+        self.experiment_xml_fp = get_output_fp('experiment.xml')
+        self.run_xml_fp = get_output_fp('run.xml')
+        self.submission_xml_fp = get_output_fp('submission.xml')
         self.pmids = s.pmids
 
-        # dicts that map investigation_type to library attributes
-        lib_strategies = {'metagenome': 'POOLCLONE',
-                          'mimarks-survey': 'AMPLICON'}
-        lib_selections = {'mimarks-survey': 'PCR'}
-        lib_sources = {}
-
-        self.library_strategy = lib_strategies.get(key, "OTHER")
-        self.library_source = lib_sources.get(key, "METAGENOMIC")
-        self.library_selection = lib_selections.get(key, "unspecified")
-
-        # This will hold the submission's samples, keyed by the sample name
+        # getting the restrictions
+        st_missing = self.sample_template.check_restrictions(
+            [self.sample_template.columns_restrictions['EBI']])
+        pt_missing = self.prep_template.check_restrictions(
+            [self.prep_template.columns_restrictions['EBI']])
+        # testing if there are any missing columns
+        if st_missing:
+            error_msgs.append(
+                "You are missing some columns in your sample template for "
+                "study #%d, preprocessed data #%d. The missing columns: %s."
+                % (s.id, preprocessed_data_id, ', '.join(list(st_missing))))
+        if pt_missing:
+            error_msgs.append(
+                "You are missing some columns in your prep template for "
+                "study #%d, preprocessed data #%d. The missing columns: %s."
+                % (s.id, preprocessed_data_id, ', '.join(list(pt_missing))))
+
```

Comment:
Its just a good coding practice.  Whenever there is redundant code, it tends to be good practice to collapse that code into helper functions - that way if there is a change in these messages, you can just modify the helper function rather than modifying many, many messages.

In this case, there are only two messages that are redundant.  However, if the number of redundant messages grows substantially, then it can be overwhelming to make any changes to these messages.

I don't know how common this error message is, so its completely your call.